guides/api-query-language-velo.md

```diff
@@ -0,0 +1,267 @@
+Sorting is applied to the first `sort` item, then the second, and so on:
+
+```json
+{
+  "sort": [
+    {
+      "fieldName": "sortField1"
+    },
+    {
+      "fieldName": "sortField2",
+      "order": "DESC"
+    }
+  ]
+}
+```
+
+## The paging object
+
+The `paging` object describes the size of the data set to return per response
+and how many records to skip.
+Each API can support **offset paging**, **cursor paging**, or both.
+See your specific API for information on supported paging options.
+
+### Offset paging
+
+With offset paging, you provide a `limit` and `offset` with each request.
+To retrieve additional pages,
+submit subsequent requests with an increased `offset`
+equal to the previous page's `limit` plus `offset`.
+
+For example, this offset request returns records 41 through 60:
+
+```json
+{
+  "paging": {
+    "limit": 20,
+    "offset": 40
+  }
+}
+```
+
+### Cursor paging
+
+With cursor paging, each request returns a `cursors` object
+that contains cursor strings that point to the next page, previous page, or both.
+To retrieve either page, use the returned `next` or `prev` cursor
+in the next request's `cursor` parameter.
+
+Take this response object, for example:
+
+```json
+{
+  "pagingMetadata": {
+    "count": 10,
+    "offset": 0,
+    "cursors": {
+      "next": "eyJmaWx0ZXIiOnsiJGFuZCI6W3sibGFuZ3VhZ2UiOnsiJGluIjpbImVuIiwiaGUiXX19LHsic3RhdHVzIjoicHVibGlzaGVkIn1dfSwidmFsdWUiOnsiaXNQaW5uZWQiOmZhbHNlLCJmaXJzdFB1Ymxpc2hlZERhdGUiOiIyMDIyLTA2LTAyVDA2OjQ2OjAyLjgwMloifSwib3JkZXIiOnsiaXNQaW5uZWQiOi0xLCJmaXJzdFB1Ymxpc2hlZERhdGUiOi0xLCJpZCI6LTF9fQ=="
+    }
+  }
+}
+```
+
+You can use the returned `next` cursor to retrieve the next page of results
+by forming your request like this:
+
+```json
+{
+  "query": {
+    "cursorPaging": {
+      "cursor": "eyJmaWx0ZXIiOnsiJGFuZCI6W3sibGFuZ3VhZ2UiOnsiJGluIjpbImVuIiwiaGUiXX19LHsic3RhdHVzIjoicHVibGlzaGVkIn1dfSwidmFsdWUiOnsiaXNQaW5uZWQiOmZhbHNlLCJmaXJzdFB1Ymxpc2hlZERhdGUiOiIyMDIyLTA2LTAyVDA2OjQ2OjAyLjgwM1oifSwib3JkZXIiOnsiaXNQaW5uZWQiOi0xLCJmaXJzdFB1Ymxpc2hlZERhdGUiOi0xLCJpZCI6LTF9fQ"
+    }
+  }
+}
+```
+
+## The fields array
+
+`fields` is an array of field paths to return.
+
+If a field path points to an object, the entire sub-object is returned.
+Subsets of sub-objects can be returned by using dot notation.
+In this example,
+the returned entities contain `firstName` from the `name` sub-object
+and the entire `address` object:
+
+```json
+{
+  "fields": [
+    "name.firstName",
+    "address"
+  ]
+}
+```
+
+## The fieldsets array
+
+An API may provide named projections to save clients from specifying individual fields in common cases.
+
+For example,
+the Contacts API implements a fieldset named `BASIC` that contains only
+`id`, `revision`, `info.name.first`, `info.name.last`,
+`primaryInfo.email`, and `primaryInfo.phone`.
+
+To use a fieldset, specify its name in the `fieldsets` array.
+
+If both `fieldsets` and `fields` arrays exist, the union of both is returned.
+For example:
```

Comment:
The code example isn't an example of the sentence that comes before. Recommend moving "for example" to its own paragraph.